src/components/store/AnalyzeStore.ts

```diff
@@ -16,44 +17,75 @@ interface AnalyzeActions {
 	}
 }
 
-// 카테고리별 뭉치고 가격 합산
-const combiningData = (
-	consumptionData: analyzeDataType[],
-): analyzeCombinedDataType[] => {
-	let id = 0
-	const consumptionCombinedData = Object.values(
-		consumptionData.reduce<Record<string, analyzeCombinedDataType>>(
-			(acc, item) => {
-				if (!acc[item.category]) {
-					acc[item.category] = {
-						id: id++,
-						category: item.category,
-						price: item.price,
-					}
-				} else {
-					acc[item.category].price += item.price
-				}
-				return acc
+function createDataItems(data: {
+	[key: string]: { value: number; firstId: number }
+}): spendingDataProps[] {
+	return Object.entries(data)
+		.sort(([, a], [, b]) => b.value - a.value)
+		.map(([key, { value, firstId }], index) => ({
+			id: firstId,
+			data: {
+				id: key,
+				label: key,
+				value,
+				color: chartColors[index % chartColors.length],
```

Comment:
혹시 여기서 `key`가 의미하는 바가 `category` 가 맞나요? 그렇다면 아래 코드처럼 `key` 를 `category`로 네이밍 해주면 id와 label, 키 값이 category 이름으로 들어가는구나가 보여서 이해하기 쉬울 것 같은데 어떠신가요? (사실 조금 헤맸거든요..)
```suggestion
function createDataItems(data: {
	[category: string]: CategorySpending
}): spendingDataProps[] {
	return Object.entries(data)
		.sort(([, a], [, b]) => b.value - a.value)
		.map(([category, { value, firstId }], index) => ({
			id: firstId,
			data: {
				id: category,
				label: category,
				value,
				color: chartColors[index % chartColors.length],
```

Reply:
제가 봐도 헷갈립니다. 전면 수정했습니다.